internal/plan/plan.go

```diff
@@ -258,6 +259,88 @@ const (
 	ActionIgnore   ServiceAction = "ignore"
 )
 
+type LogTargets struct {
+	Targets []string
+	Keyword string
+}
+
+const (
+	// Valid keywords for LogTargets
+	DefaultLogTargets = "default"
+	AllLogTargets     = "all"
+	NoLogTargets      = "none"
+)
+
+func (t *LogTargets) UnmarshalYAML(value *yaml.Node) error {
+	if value.Kind == yaml.ScalarNode {
+		switch value.Value {
+		case DefaultLogTargets, AllLogTargets, NoLogTargets:
+			t.Keyword = value.Value
+			return nil
+		default:
+			return fmt.Errorf("invalid value %q for LogTargets", value.Value)
+		}
+	}
+
+	// unmarshal to []string
+	return value.Decode(&t.Targets)
+}
+
+// Copy returns a deep copy of this LogTargets struct.
+func (t *LogTargets) Copy() *LogTargets {
+	return &LogTargets{
+		Targets: append([]string(nil), t.Targets...),
+		Keyword: t.Keyword,
+	}
+}
+
+// Merge merges the fields from other into t.
+func (t *LogTargets) Merge(other *LogTargets) {
+	// anything <- keyword yields keyword
+	if other.Keyword != "" {
+		t.Keyword = other.Keyword
+		t.Targets = nil
+	}
+
+	// keyword <- [list] yields [list]
+	if t.Keyword != "" && len(other.Targets) > 0 {
+		t.Keyword = ""
+		t.Targets = append([]string(nil), other.Targets...)
+		return
+	}
+
+	// [list] <- [list] - merge lists
+	t.Targets = appendUnique(t.Targets, other.Targets...)
+}
+
+func (t *LogTargets) LogsTo(tgt *LogTarget) bool {
+	// Handle keywords
+	switch t.Keyword {
+	case AllLogTargets:
+		return tgt.Selection != DisabledSelection
+	case DefaultLogTargets:
+		return tgt.Selection == OptOutSelection || tgt.Selection == UnsetSelection
+	case NoLogTargets:
+		return false
+	}
+
+	// No keyword
+	if tgt.Selection == DisabledSelection {
```

Comment:
I've not spent too long looking at this, but if feels like some code duplication can be removed if we merge some of the checks?

```
if tgt.Selection == DisabledSelection ||  t.Keyword == NoLogTargets {
    return false
}
```
As the first check?

Reply:
I was thinking of merging all the case statements with the if/else cases below, similar to the one example I provided. But to be honest I have not thought it through completely, so maybe the outcome does not look much nicer.

Reply:
@flotter I updated the method, let me know what you think.